### ISSUE TITLE
Make an d3-area-chunked variant

### ISSUE THREAD

**Issue by fvanwijk** [pbeshai/d3-line-chunked#16]
Peter, your line function is awesome. It solves a use case for us that otherwise would take hours of work to implement ourselves.

We also have a lot of areas, for which we want to have the same behaviour as with the line-chunked function. We could fork this repo to adjust the function but maybe you will soon find the time to release an area-variant of this function.

**Comment by pbeshai** [comment]
Hi @fvanwijk, thanks! I don't think I'll have time to do this any time soon. Good luck putting it together, would love to see what you come up with!

**Comment by fvanwijk** [comment]
After a couple of hours hacking, this is what I created: https://github.com/fvanwijk/d3-area-chunked
Looks familiar right? ;)

I didn't need to change too much code to make it work.

1. Rename `line` to `area` in code, comments, readme, etc.
2. Use `d3.area` instead of `d3.line`
3. Set the `y0` attribute to y=0. `y` is renamed to `y1`.
4. Adjust clipping areas to span the full y range

Since I reused (copied) 98% of your code, I wonder if I could build in the `area` part into this repo. I can imagine that the 'line' would render as 'area' - or both - using a parameter.
However, this way the name `d3-line-chunked` doesn't fit anymore.
An alternative approach would be to extract all generic code and create a `line` and `area` constructor that uses the generic code for rendering.

On the other hand we could leave the `d3-area-chunked` in a separate codebase, giving me freedom to add/change features and to move away from the existing codebase.
